Source/AwsCommonRuntimeKit/auth/credentials/CRTAWSCredentialsProvider.swift

```diff
@@ -57,19 +115,21 @@ public final class CRTAWSCredentialsProvider {
             shutdownCallbackCore.release()
             throw CommonRunTimeError.crtError(CRTError.makeFromLastError())
         }
-        self.init(credentialsProvider: provider, allocator: allocator)
+        return AwsCredentialsProvider(credentialsProvider: provider, allocator: allocator)
     }
 
-    /// Creates a credentials provider from environment variables:
+    /// Creates a credentials provider that returns credentials based on environment variable values:
     /// - `AWS_ACCESS_KEY_ID`
     /// - `AWS_SECRET_ACCESS_KEY`
     /// - `AWS_SESSION_TOKEN`
     ///
     /// - Parameters:
-    ///   - shutdownOptions:  The `CredentialsProviderShutdownOptions`options object.
-    /// - Returns: `AWSCredentialsProvider`
-    public convenience init(fromEnv shutdownCallback: ShutdownCallback?,
-                            allocator: Allocator = defaultAllocator) throws {
+    ///   - shutdownCallback:  (Optional) shutdown callback.
+    ///   - allocator: (Optional) allocator to override.
+    /// - Returns: `CredentialsProvider`
+    /// - Throws: CommonRuntimeError.crtError
+    public static func makeEnvironment(shutdownCallback: ShutdownCallback? = nil,
+                                       allocator: Allocator = defaultAllocator) throws -> AwsCredentialsProvider {
```

Comment:
Similar to a comment above, it's preferable to move this to an initializer rather than a static function.